### PR TITLE
Icc python interface + Icc core fixes

### PR DESCRIPTION
**Pull request by KonradBreitsprecher**

### CODE REVIEW

**Comment by fweik** [comment]
@KonradBreitsprecher could you please merge with the master. I'm not sure if I understand that correctly. This collides with the change of the verlet criterion by @RudolfWeeber

**Comment by fweik** [comment]
Bump

File: .travis.yml [hunk 1]
```diff
@@ -65,6 +64,3 @@ compiler:
   - clang
 
 script: ./maintainer/travis/build_cmake.sh
-
-after_success:
- - codecov
```
Comment (KaiSzuttor) [inline .travis.yml]: Why is this file changed?

File: samples/python/p3m.py [hunk 2]
```diff
@@ -115,10 +115,22 @@
...
 system.actors.add(p3m)
 
-print("P3M parameter:\n")
+print("\nSCRIPT--->P3M parameter:\n")
+p3m_params = p3m.get_params()
+for key in p3m_params.keys():
+    print("{} = {}".format(key, p3m_params[key]))
+
+print("\nSCRIPT--->Explicit tune call\n")
+p3m._tune()
```
Comment (KaiSzuttor) [inline samples/python/p3m.py]: Maybe there should be a method of the P3M class to call the tune routine instead of using the as "internal" marked `_tune` method.

**Comment by fweik** [comment]
Looks ok to me. But agree with @KaiSzuttor that there should be a public tune method. Can you add that? Otherwise I would merge.

**Comment by KonradBreitsprecher** [comment]
Yes @KaiSzuttor has a point there. Will add this next time.
